contrib/pyln-testing/pyln/testing/utils.py

```diff
@@ -1158,10 +1158,20 @@ def wait_for_htlcs(self, scids=None):
                     wait_for(lambda: len(self.rpc.listpeerchannels(peer["id"])['channels'][idx]['htlcs']) == 0)
 
     # This sends money to a directly connected peer
-    def pay(self, dst, amt, label=None):
+    # if `route` is `True`, it can also send over the network.
+    def pay(self, dst, amt, label=None, route=False):
```

Comment:
It could be very cool to be able to specify a `route` with specific hops such that we can constrain where the payment goes through. In that case I'd say `route=None` only uses peers, `multihop=True` means using `getroute` to find a path, and `route=[...]` would mean we constrain the payment to the specific route.

Just an idea to keep naming options in mind for future improvements.

Reply:
Good idea, let me check...